### PR TITLE
Fixed back button hiding with restored lastPage

### PR DESCRIPTION
**Pull request by MightyPork**
This PR fixes a bug introduced by remembering the last page and restoring to it on startup.
The Back button now goes to the home page ("Listen Now") if there's no history to use.

I also took the liberty to clean up the file and split it to functions, I hope you're okay with that. Better than having all the tweaks in a single block.

### CODE REVIEW

**Comment by MightyPork** [comment]
uh I may have broken something. Wait a bit, will make sure this is right.

File: src/inject/GPMInject/interface/customUI.js [hunk 1]
```diff
@@ -71,14 +90,29 @@ window.wait(() => {
...
+  const listenNowURL = 'https://play.google.com/music/listen#/now';
+
+  const backBtn = document.createElement('paper-icon-button');
+  backBtn.setAttribute('icon', 'arrow-back');
+  backBtn.setAttribute('id', 'backButton');
+  backBtn.setAttribute('class', 'x-scope paper-icon-button-0');
+  document.querySelector('#material-one-middle > sj-search-box').insertBefore(backBtn, null);
+
+  backBtn.addEventListener('click', () => {
+    const oldUrl = location.href;
```
Comment (MarshallOfSound) [inline src/inject/GPMInject/interface/customUI.js]: I've got a better idea for this code

```js
remote.getCurrentWindow().webContents.executeJavaScript('document.querySelector("webview").canGoBack()', false, (canGoBack) => {
    if (canGoBack) return history.back();
    location.href = listenNowUrl;
});
```

This is less vulnerable code (a settimeout relies on PC speed and network speed), this uses proper electron API's :+1:

**Comment by MightyPork** [comment]
Update: Looks good. It was a bug on my home-use branch. Master is good.

File: src/inject/GPMInject/interface/customUI.js (hunk 1 above)
Comment (MightyPork) [inline src/inject/GPMInject/interface/customUI.js]: I'll give that a go. Certainly more elegant :+1:

Comment (MightyPork) [inline src/inject/GPMInject/interface/customUI.js]: Works like a charm!

Comment (MarshallOfSound) [inline src/inject/GPMInject/interface/customUI.js]: Considering I wrote that on my phone while walking home I am quite impressed it just worked :+1:

**Comment by MarshallOfSound** [comment]
Can we start getting into the habbit of squishing PR's into single commits.  It will make the history look a lot cleaner

**Comment by MightyPork** [comment]
Right, done. I usually do that, but didn't think it's needed here. 

Ready for merge, if Travis doesn't make a scene.

**Pull request merged**